typescript/infra/scripts/helloworld/kathy.ts

```diff
@@ -114,11 +135,28 @@ async function sendMessage(
     body: msg,
   };
   const value = await gasCalc.estimatePaymentForMessage(expected);
-  console.log(`Sending message from ${origin} to ${destination}`);
-  const receipt = await app.sendHelloWorld(origin, destination, msg, value);
-  console.log(JSON.stringify(receipt.events || receipt.logs));
+
+  log('Sending message', { from: origin, to: destination });
+
+  await new Promise<ethers.ContractReceipt[]>((resolve, reject) => {
+    setTimeout(
+      () => reject(new Error('Timeout waiting for message receipt')),
+      10 * 60 * 1000,
+    );
+    app
+      .sendHelloWorld(origin, destination, msg, value, (receipt) => {
+        log('Message sent', {
+          from: origin,
+          to: destination,
+          events: receipt.events,
+          logs: receipt.logs,
+        });
+      })
+      .then(resolve)
+      .catch(reject);
+  });
```

Comment:
is this different from 
```suggestion
    setTimeout(
      () => throw new Error('Timeout waiting for message receipt')
      10 * 60 * 1000,
    );
    await app
      .sendHelloWorld(origin, destination, msg, value, (receipt) => {
        log('Message sent', {
          from: origin,
          to: destination,
          events: receipt.events,
          logs: receipt.logs,
        });
      });
```

Reply:
I thought that if you throw an error like you are it would not cause the current function to stop awaiting the promise but just raise the error externally, like when you don't catch a promise.

Reply:
cc @nambrot for promisification